Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java

```diff
@@ -554,6 +558,28 @@ private void removeRepoFromCache(String authority) {
             if (notEmpty.contains(uri)) continue;
             deleteSeries(context, uri);
         }
+        
+        if (recurse) {
```

Comment:
this code appears to be copy and pasted across all the types.  we should refactor this out into a common class and you can pass in an interface that has the methods:

getUriMap(String uri, int depth) <-- this would replace the list*byUriPrefix methods based on what method you're in
deleteUri(String uri) <-- this would replace the deletion operation when the size of the map is 0